docker-entrypoint.sh

```diff
@@ -73,6 +76,17 @@ if [[ "$1" == "lnd" || "$1" == "lncli" ]]; then
         echo "externalip=$HIDDENSERVICE_ONION added to $LND_DATA/lnd.conf"
     fi
 
+    # if it is legacy installation, then trigger warning and add noseedbackup=1 to config if needed
+    WALLET_FILE="$LND_DATA/data/chain/$NETWORK/$ENV/wallet.db"
+    LNDUNLOCK_FILE=${WALLET_FILE/wallet.db/walletunlock.json}
+    if [ -f "$WALLET_FILE" -a  ! -f "$LNDUNLOCK_FILE" ]; then
+        echo "[lnd_unlock] WARNING: UNLOCK FILE DOESN'T EXIST! MIGRATE LEGACY INSTALLATION TO NEW VERSION ASAP"
+        echo "noseedbackup=1" >> "$LND_DATA/lnd.conf"
+    else
+    # else hit up the auto initializer and unlocker to do it's work
+        nohup ./docker-initunlocklnd.sh $NETWORK $ENV &
```

Comment:
I think you should remove `nohup` you just want `initunlocklnd.sh` to be executed in parallel.

Reply:
I'm not bash expert... but did this on purpose because I needed a new process. You're saying that without nohup `docker-initunlocklnd.sh` will still spawn new process and continue execution bellow to start LND?

Reply:
Yes, `SIGHUP` is the name of a linux signal sent to an app when the tty (the interactive bash session which spawned the process) close. By default SIGHUP is handled by apps and interpreted like a kill. 

`nohup blahblah`  means that process blahblah won't receive the signal, because the signal go to the `nohup` process which just drop it. So `blahblah` does not close.

`nohup blahblah &` means "start blablah on the background, and don't close it if I leave the terminal".

But here,
1. There is no terminal
2. If there was a terminal, you want the background process to close.

Reply:
Thanks for detailed explanation.

My worry was focused on that just doing `./docker-initunlocklnd.sh $NETWORK $ENV` would block the execution of other commands in the script. After more reading on bash I now understand that appending that `&` at the end will ensure that `docker-initunlocklnd.sh` is started and that `docker-entrypoint.sh` will continue with execution... and you've provided rest of knowledge with explaination of `nohup`

See fe95b53af50864272232f42a770c2a8e731d228e